qualtran/linalg/permutation.py

```diff
@@ -47,7 +47,7 @@ def decompose_permutation_map_into_cycles(permutation_map: dict[int, int]) -> It
 
         # compute the cycle starting at `i`
         cycle = []
-        while True:
+        while i not in seen:
```

Comment:
add a test that would have failed?

Reply:
added in https://github.com/quantumlib/Qualtran/pull/1307